From d563d17154b3bbff96d0e2cd39ba69dd6ef2822e Mon Sep 17 00:00:00 2001
From: Alex Naidis <alex.naidis@linux.com>
Date: Tue, 4 Oct 2016 14:08:33 +0200
Subject: [PATCH 3/3] jemalloc: don't query /proc for the overcommit setting

* @arter97 found the issue that
  querying /proc to get the OS's
  overcommit policy at init time,
  creates an overhead.
-> Hardcode the overcommit policy
   for Android, since Android always
   does overcommit, in order to avoid
   retrieving this information via
   /proc.

Change-Id: Ibb1acad2d939ccbbadb5566c00186137980838e6
Signed-off-by: Alex Naidis <alex.naidis@linux.com>
---
 src/pages.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/pages.c b/src/pages.c
index 51d55f5..f08827b 100644
--- a/src/pages.c
+++ b/src/pages.c
@@ -13,7 +13,11 @@
 #  define PAGES_PROT_DECOMMIT (PROT_NONE)
 static int	mmap_flags;
 #endif
+#if !defined(__ANDROID__)
 static bool	os_overcommits;
+#else
+static const bool	os_overcommits = true;
+#endif
 
 /******************************************************************************/
 /* Defines/includes needed for special android code. */
@@ -253,11 +257,11 @@ os_overcommits_proc(void)
 void
 pages_boot(void)
 {
-
 #ifndef _WIN32
 	mmap_flags = MAP_PRIVATE | MAP_ANON;
 #endif
 
+#if !defined(__ANDROID__)
 #ifdef JEMALLOC_SYSCTL_VM_OVERCOMMIT
 	os_overcommits = os_overcommits_sysctl();
 #elif defined(JEMALLOC_PROC_SYS_VM_OVERCOMMIT_MEMORY)
@@ -269,4 +273,5 @@ pages_boot(void)
 #else
 	os_overcommits = false;
 #endif
+#endif
 }
-- 
2.10.0

